ProfileSample/Controllers/HomeController.cs

```diff
@@ -47,18 +44,20 @@ public ActionResult Convert()
                     {
                         byte[] buff = new byte[stream.Length];
 
-                        stream.Read(buff, 0, (int) stream.Length);
+                        stream.ReadAsync(buff, 0, (int)stream.Length);
```

Comment:
This won't work without `await` 😃 I was about making the whole action and calls to external resources asynchronous

https://learn.microsoft.com/en-us/aspnet/core/performance/performance-best-practices?view=aspnetcore-7.0#avoid-blocking-calls